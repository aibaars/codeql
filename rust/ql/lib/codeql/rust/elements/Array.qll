// generated by codegen, remove this comment if you wish to edit this file
/**
 * This module provides a hand-modifiable wrapper around the generated class `Array`.
 */

private import codeql.rust.generated.Array

class Array extends Generated::Array { }
