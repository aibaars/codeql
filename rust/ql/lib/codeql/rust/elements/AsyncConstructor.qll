// generated by codegen, remove this comment if you wish to edit this file
/**
 *  This module defines the hook used internally to tweak the characteristic predicate of
 *  `Async` synthesized instances.
 *  INTERNAL: Do not use.
 */

private import codeql.rust.generated.Raw

/**
 *  The characteristic predicate of `Async` synthesized instances.
 *  INTERNAL: Do not use.
 */
predicate constructAsync(Raw::Async id) { any() }
