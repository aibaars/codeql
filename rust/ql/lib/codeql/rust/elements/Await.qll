// generated by codegen, remove this comment if you wish to edit this file
/**
 * This module provides a hand-modifiable wrapper around the generated class `Await`.
 */

private import codeql.rust.generated.Await

class Await extends Generated::Await { }
