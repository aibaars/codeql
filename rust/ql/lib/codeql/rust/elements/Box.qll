// generated by codegen, remove this comment if you wish to edit this file
/**
 * This module provides a hand-modifiable wrapper around the generated class `Box`.
 */

private import codeql.rust.generated.Box

class Box extends Generated::Box { }
