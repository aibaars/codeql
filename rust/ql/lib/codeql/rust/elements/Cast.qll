// generated by codegen, remove this comment if you wish to edit this file
/**
 * This module provides a hand-modifiable wrapper around the generated class `Cast`.
 */

private import codeql.rust.generated.Cast

class Cast extends Generated::Cast { }
