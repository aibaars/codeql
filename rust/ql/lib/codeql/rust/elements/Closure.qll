// generated by codegen, remove this comment if you wish to edit this file
/**
 * This module provides a hand-modifiable wrapper around the generated class `Closure`.
 */

private import codeql.rust.generated.Closure

class Closure extends Generated::Closure { }
