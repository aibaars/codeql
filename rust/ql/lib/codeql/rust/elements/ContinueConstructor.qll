// generated by codegen, remove this comment if you wish to edit this file
/**
 *  This module defines the hook used internally to tweak the characteristic predicate of
 *  `Continue` synthesized instances.
 *  INTERNAL: Do not use.
 */

private import codeql.rust.generated.Raw

/**
 *  The characteristic predicate of `Continue` synthesized instances.
 *  INTERNAL: Do not use.
 */
predicate constructContinue(Raw::Continue id) { any() }
