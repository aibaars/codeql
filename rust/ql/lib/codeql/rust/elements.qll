// generated by codegen
/**
 * This module exports all modules providing `Element` subclasses.
 */

import codeql.rust.elements.Array
import codeql.rust.elements.Async
import codeql.rust.elements.Await
import codeql.rust.elements.Become
import codeql.rust.elements.BinaryOp
import codeql.rust.elements.BindPat
import codeql.rust.elements.Block
import codeql.rust.elements.Box
import codeql.rust.elements.BoxPat
import codeql.rust.elements.Break
import codeql.rust.elements.Call
import codeql.rust.elements.Cast
import codeql.rust.elements.Closure
import codeql.rust.elements.Const
import codeql.rust.elements.ConstBlockPat
import codeql.rust.elements.Continue
import codeql.rust.elements.DbFile
import codeql.rust.elements.DbLocation
import codeql.rust.elements.Declaration
import codeql.rust.elements.Element
import codeql.rust.elements.Expr
import codeql.rust.elements.ExprStmt
import codeql.rust.elements.Field
import codeql.rust.elements.File
import codeql.rust.elements.Function
import codeql.rust.elements.If
import codeql.rust.elements.IfLet
import codeql.rust.elements.Index
import codeql.rust.elements.InlineAsm
import codeql.rust.elements.ItemStmt
import codeql.rust.elements.Let
import codeql.rust.elements.LitPat
import codeql.rust.elements.Literal
import codeql.rust.elements.Locatable
import codeql.rust.elements.Location
import codeql.rust.elements.Loop
import codeql.rust.elements.Match
import codeql.rust.elements.MatchArm
import codeql.rust.elements.MethodCall
import codeql.rust.elements.MissingExpr
import codeql.rust.elements.MissingPat
import codeql.rust.elements.Module
import codeql.rust.elements.OffsetOf
import codeql.rust.elements.OrPat
import codeql.rust.elements.Pat
import codeql.rust.elements.Path
import codeql.rust.elements.PathPat
import codeql.rust.elements.Range
import codeql.rust.elements.RangePat
import codeql.rust.elements.RecordLit
import codeql.rust.elements.RecordPat
import codeql.rust.elements.Ref
import codeql.rust.elements.RefPat
import codeql.rust.elements.Return
import codeql.rust.elements.SlicePat
import codeql.rust.elements.Stmt
import codeql.rust.elements.Tuple
import codeql.rust.elements.TuplePat
import codeql.rust.elements.TupleStructPat
import codeql.rust.elements.TypeRef
import codeql.rust.elements.UnaryExpr
import codeql.rust.elements.Underscore
import codeql.rust.elements.UnknownFile
import codeql.rust.elements.UnknownLocation
import codeql.rust.elements.Unsafe
import codeql.rust.elements.WildPat
import codeql.rust.elements.Yeet
import codeql.rust.elements.Yield
