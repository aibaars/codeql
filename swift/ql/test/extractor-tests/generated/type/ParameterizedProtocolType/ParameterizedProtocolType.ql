// generated by codegen/codegen.py
import codeql.swift.elements
import TestUtils

from ParameterizedProtocolType x, string getName, Type getCanonicalType, ProtocolType getBase
where
  toBeTested(x) and
  not x.isUnknown() and
  getName = x.getName() and
  getCanonicalType = x.getCanonicalType() and
  getBase = x.getBase()
select x, "getName:", getName, "getCanonicalType:", getCanonicalType, "getBase:", getBase
