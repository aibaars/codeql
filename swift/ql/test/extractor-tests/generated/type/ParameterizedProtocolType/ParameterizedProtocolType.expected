| P<Int, String> | getName: | P<Int, String> | getCanonicalType: | P<Int, String> | getBase: | P |
